frontend/src/components/DefenceBox/DefenceMechanism.tsx

```diff
@@ -7,11 +7,13 @@ const { TiTick, TiTimes } = require("react-icons/ti");
 
 function DefenceMechanism({
   defenceDetail,
+  currentPhase,
   setDefenceActive,
   setDefenceInactive,
   setDefenceConfiguration,
 }: {
   defenceDetail: DefenceInfo;
+  currentPhase: number;
```

Comment:
I'd personally prefer this to be a boolean called something like `showConfiguration`. That way the component doesn't need to know about phases.

Reply:
sure can do that